app/models/competition.rb

```diff
@@ -925,6 +936,8 @@ def receive_registration_emails=(r)
     update_column :external_website, nil
   end
 
+  after_save :auto_close, if: -> { saved_change_to_auto_close_threshold? }
```

Comment:
Super-duper unhappy about `after_save`s in our huge competition god model. Is this absolutely necessary for the feature to work?

As I understand it, checking in the `after_create` of the competition payments should be enough.

Reply:
Also, if we decide to keep it: `if: :saved_change_to_auto_close_threshold?` should suffice.

Reply:
The intention here is to prevent the organizer changing the threshold to/below the number of paid registrations - but I realize that this should just be a model validation instead.

Reply:
Validation added